File: Data Structures & Algorithms/find-duplicate-integer/submission-6.java
class Solution {
    public int findDuplicate(int[] nums) {
        boolean[] seen = new boolean[nums.length];

        for (int num : nums) {
            if (seen[num - 1]) {
                return num;
            }
            seen[num - 1] = true;
        }
        return -1;
    }
}

File: Data Structures & Algorithms/subsets/submission-6.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = [[]]

        for num in nums:
            for i in range(len(res)):
                subset = res[i][:]
                subset.append(num)
                res.append(subset)
            
        
        return res 